src/transactions/test/SetTrustLineFlagsTests.cpp

```diff
@@ -287,3 +288,767 @@ TEST_CASE("set trustline flags", "[tx][settrustlineflags]")
+                        }
+                    };
+
+                auto submitRevokeInSandwich = [&](TestAccount& sponsoringAcc,
+                                                  TestAccount& sponsoredAcc,
+                                                  bool success) {
+                    auto op =
+                        flagOp == TrustFlagOp::ALLOW_TRUST
+                            ? allowTrust(acc1, cur1, 0)
+                            : setTrustLineFlags(
+                                  acc1, cur1,
+                                  clearTrustLineFlags(TRUSTLINE_AUTH_FLAGS));
+
+                    auto tx = transactionFrameFromOps(
+                        app->getNetworkID(), sponsoringAcc,
+                        {sponsoringAcc.op(
+                             beginSponsoringFutureReserves(sponsoredAcc)),
+                         root.op(op),
+                         sponsoredAcc.op(endSponsoringFutureReserves())},
+                        {sponsoredAcc, root});
+
+                    {
+                        LedgerTxn ltx(app->getLedgerTxnRoot());
+                        TransactionMeta txm(2);
+                        REQUIRE(tx->checkValid(ltx, 0, 0, 0));
+                        REQUIRE(tx->apply(*app, ltx, txm) == success);
+                        ltx.commit();
+                    }
+
+                    if (success)
+                    {
+                        REQUIRE(tx->getResultCode() == txSUCCESS);
+                    }
+                    else
+                    {
+                        REQUIRE(tx->getResultCode() == txFAILED);
+                        if (flagOp == TrustFlagOp::ALLOW_TRUST)
+                        {
+                            REQUIRE(tx->getResult()
+                                        .result.results()[1]
+                                        .tr()
+                                        .allowTrustResult()
+                                        .code() == ALLOW_TRUST_LOW_RESERVE);
+                        }
+                        else
+                        {
+                            REQUIRE(tx->getResult()
+                                        .result.results()[1]
+                                        .tr()
+                                        .setTrustLineFlagsResult()
+                                        .code() ==
+                                    SET_TRUST_LINE_FLAGS_LOW_RESERVE);
+                        }
+                    }
+                };
+
+                auto increaseReserve = [&]() {
+                    // double the reserve
+                    auto newReserve = lm.getLastReserve() * 2;
+                    REQUIRE(
+                        executeUpgrade(*app, makeBaseReserveUpgrade(newReserve))
+                            .baseReserve == newReserve);
+                };
+
+                // same reserves
+                SECTION("same reserve - no sandwich on revoke")
```

Comment:
I like this. Tests like this are essentially what I had in mind in my long comment above.